### ISSUE TITLE
cartopy coastlines issue with shapely 1.5.13

### ISSUE THREAD

**Issue by dluzenst** [SciTools/cartopy#682]
With shapely 1.5.13 the following cartopy example generates an incomplete figure:

```
import os
import matplotlib.pyplot as plt

from cartopy import config
import cartopy.crs as ccrs


fig = plt.figure(figsize=(8, 12))

# get the path of the file. It can be found in the repo data directory.
fname = os.path.join(config["repo_data_dir"],
                     'raster', 'sample', 'Miriam.A2012270.2050.2km.jpg'
                     )
img_extent = (-120.67660000000001, -106.32104523100001, 13.2301484511245, 30.766899999999502)
img = plt.imread(fname)

ax = plt.axes(projection=ccrs.PlateCarree())
plt.title('Hurricane Miriam from the Aqua/MODIS satellite\n'
          '2012 09/26/2012 20:50 UTC')

# set a margin around the data
ax.set_xmargin(0.05)
ax.set_ymargin(0.10)

# add the image. Because this image was a tif, the "origin" of the image is in the
# upper left corner
ax.imshow(img, origin='upper', extent=img_extent, transform=ccrs.PlateCarree())
ax.coastlines(resolution='50m', color='black', linewidth=1)

# mark a known place to help us geo-locate ourselves
ax.plot(-117.1625, 32.715, 'bo', markersize=7, transform=ccrs.Geodetic())
ax.text(-117, 33, 'San Diego', transform=ccrs.Geodetic())

plt.show()
```

I think the error comes from the ax.coastlines line. The error goes away by reverting back to shapely 1.5.12. 
Otherwise I'm using cartopy 0.13.0 and matplotlib 1.4.3 on CentOS 6.
The code comes from http://scitools.org.uk/cartopy/docs/latest/matplotlib/advanced_plotting.html.

Thought this might be useful to know.

**Comment by ocefpaf** [comment]
I can confirm that.  Cartopy/Matplotlib crashes with `TypeError: unhashable type: 'Polygon'`

I guess that the culprit is:

https://github.com/Toblerity/Shapely/commit/8df2b1b718c89e7d644b246ab07ad3670d25aa6a

Ping @rsignell-usgs who is having this issue with some of his [notebooks](https://raw.githubusercontent.com/rsignell-usgs/notebook/master/WFS/WFS_plot_in_Cartopy.ipynb).

**Comment by ocefpaf** [comment]
I narrowed it down to the `weakref.WeakKeyDictionary()`  [`_geometry_to_path_cache`](https://github.com/SciTools/cartopy/blob/b2f8fd38d1ecba73a9c1dc40807293a7ce7766e9/lib/cartopy/mpl/feature_artist.py#L40)

Now that Shapely polygons are no longer hashable we cannot do:

```python
mapping = FeatureArtist._geometry_to_path_cache.setdefault(geom, {})
```

**Comment by sgillies** [comment]
Sorry about the breakage, folks. Making Shapely geoms hashable was a mistake, but in hindsight the fix might better have been left to Shapely 1.6.

**Comment by rhattersley** [comment]
A workaround shouldn't be hard - I'll post something soon.
